presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -207,7 +209,7 @@ public PlanWithProperties visitGroupId(GroupIdNode node, HashComputationSet pare
         public PlanWithProperties visitDistinctLimit(DistinctLimitNode node, HashComputationSet parentPreference)
         {
             // skip hash variable generation for single bigint
-            if (canSkipHashGeneration(node.getDistinctVariables())) {
+            if (canSkipHashGeneration(node.getDistinctVariables()) || SystemSessionProperties.isRoundRobinShuffleBeforePartialAgg(session)) {
```

Comment:
Can you explain what's going on with skipping hash generation?

(Also should match on the plan here rather than session properties - but if this code stays, should be static import)